service/src/test/java/dk/dbc/search/solrdocstore/BibliographicBeanIT.java

```diff
@@ -102,11 +106,181 @@ public void newCommonRecordWithExistingHoldings() throws Exception {
 
         l = em.createQuery("SELECT h FROM HoldingsToBibliographicEntity as h WHERE h.bibliographicRecordId='new'", HoldingsToBibliographicEntity.class).getResultList();
         assertThat(l, containsInAnyOrder(
-                   new HoldingsToBibliographicEntity(700000, "new", 700000),
-                   new HoldingsToBibliographicEntity(700100, "new", 870970),
-                   new HoldingsToBibliographicEntity(300100, "new", 300100),
-                   new HoldingsToBibliographicEntity(300200, "new", 300000)
-           ));
+                new HoldingsToBibliographicEntity(700000, "new", 700000),
+                new HoldingsToBibliographicEntity(700100, "new", 870970),
+                new HoldingsToBibliographicEntity(300100, "new", 300100),
+                new HoldingsToBibliographicEntity(300200, "new", 300000)
+        ));
+    }
+
+    @Test
+    public void updateExistingBibliographicPost() throws JSONBException {
+        BibliographicEntity b = new BibliographicEntity(600100,"properUpdate","work:update","unit:update","prod:update",false,new HashMap<>(),"track:update");
+        String updatedB = jsonbContext.marshall(b);
+
+        Response r = env().getPersistenceContext()
+                .run(() -> bean.addBibliographicKeys(null, updatedB)
+                );
+        assertThat(r.getStatus(), is(200));
+
+        // Ensure update came through
+        BibliographicEntity updatedBibEntity = em.find(BibliographicEntity.class, new AgencyItemKey(b.agencyId, b.bibliographicRecordId));
+        assertThat(updatedBibEntity,equalTo(b));
+        //assertThat(true,equalTo(true));
+
+        // Ensure related holdings are unchanged
+        List<HoldingsToBibliographicEntity> l = em.createQuery("SELECT h FROM HoldingsToBibliographicEntity h WHERE h.bibliographicRecordId='properUpdate'", HoldingsToBibliographicEntity.class).getResultList();
+        assertThat(l,containsInAnyOrder(
+                new HoldingsToBibliographicEntity(610510,"properUpdate",600100)
+        ));
+    }
+
+    /**
+     * If a bibliographic post (all types) goes from deleted: false -> true, that bibliographic holdings are properly updated
+     * @throws JSONBException
+     */
+    @Test
+    public void deleteUpdateHoldings() throws JSONBException {
+        // Update common for FBS
```

Comment:
Lav et "før"-billede (assertThat ....) eller skal man over i "bibliographicUpdate.sql" for at finde før billedet.
Det bliver bare lettere at læse testen enkeltstående.